File: backend/GrTabStore/Models/Ad.cs
using System.ComponentModel.DataAnnotations;
namespace GraphicsTabletStore.API.Models
{
    public class Ad
    {
        public int AdId { get; set; }
        public string Title { get; set;} = string.Empty;
        public required string ImageUrl { get; set;}
        public string? Link { get; set;} = string.Empty;

        public bool? IsDark { get; set; } = false;

    }
}

File: frontend/lib/data/models/ad.dart
class Ad {
  final int id;
  final String title;
  final String? imageUrl;
  final String? link;
  final bool isDark;

  Ad({
    required this.id,
    required this.title,
    this.imageUrl,
    this.link,
    this.isDark = false,
  });

  factory Ad.fromJson(Map<String, dynamic> json) => Ad(
    id: int.tryParse(json['id'].toString()) ?? 0,
    title: (json['title'] as String?) ?? '',
    imageUrl: json['imageUrl'] as String?,
    link: json['link'] as String?,
    isDark: (json['isDark'] as bool?) ?? false,
  );

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'imageUrl': imageUrl,
      'link': link,
      'isDark': isDark,
    };
  }
}
